### DISCUSSION TITLE
Project Showcase: hive

### DISCUSSION THREAD

**Discussion by ghost** [TomSchimansky/CustomTkinter#1731] [Showcase]
### Introduction:
Hi everyone! For those who don't know, I'm Dishant :) I'm new to open-source (kind-of?) and I'm working on a project called [`hive`](https://github.com/dishb/hive).

As you've probably already guessed, `hive` is developed with Python and `customtkinter`. I'm currently workign on this project solo, but new contributors are more than welcome! I'll love to see this project gain traction and be used.

### About:
`hive` is a simple file explorer (only for macOS right now). It's targeted at casual users - people who don't need the full power of the file system. People who only need to see their downloads and files. `hive` is currently in heavy development. It currently allows users to navigate through the file system, view "system" (otherwise hidden) files and directories, open files, customize themes, and other UI-related things!

It's quite simple but I hope to add renaming and moving files. After that - I might be ready for a release! Though you can build `hive` from source right now.

### Images:
![1](https://github.com/TomSchimansky/CustomTkinter/assets/130259005/c4065601-d885-4d6b-9bea-f1cc758cebab)
![2](https://github.com/TomSchimansky/CustomTkinter/assets/130259005/611ff91b-a4d6-4c97-84b0-ea27699b4237)
![3](https://github.com/TomSchimansky/CustomTkinter/assets/130259005/be99951a-3b1b-44ab-8a07-7f7c51444a0a)
![4](https://github.com/TomSchimansky/CustomTkinter/assets/130259005/d9beb468-328d-4355-b50d-5fbc157ab031)

### Contributing:
To attract new contributors, I've tried to make the process as simple as it can be. I've left clear, detailed instructions in the README.md file for building and contributing to `hive`. I've also used a commit style-guide to keep development organized.

I made this post to share my project and hopefully attract new users and developers to the project. That's all I really had to say. I might add some images of `hive` soon! :)

### Final notes:
- If you have any ideas, please open a Discussion in the Ideas section! I'd love to hear what you have to say :)
- For bugs and issues, please open an Issue on Github.
- For security vulnerabilities, please privately [email](mailto:code.dishb@gmail.com) me.

**Comment by avalon60** [comment]
Hi @dishb. That link doesn't seem to point to anything like what you are describing?

**Reply by ghost** [reply]
@avalon60 - Oh that's embarrassing- I forgot to add my username in the URL. It is fixed now.

**Comment by avalon60** [comment]
@dishb,  If I were you, I would set up a Discussions section for the repo. 

By the way, I would consider changing the installation instructions, so that they don't prescribe configuring the modules globally, which can be problematic and lead to a bloated Python environment. 
Using virtual environments addresses this, and also prevents module version conflicts (and other apps breaking), where you are working on multiple projects. 

You can avoid this with something like:

```
mkdir <project_location>
cd <project_location>
python3 -m venv venv       # create a virtual environment and place it in a sub-directory called venv (the second venv is the directory name)
source venv/bin/activate    # Activate the virtual environment
python3 get-pip.py

pip3 install --upgrade requests
pip3 install --upgrade pyinstaller
pip3 install --upgrade customtkinter
pip3 install --upgrade pillow
```

You can also eliminate the 4 pip3 install lines, once you have your own virtual environment initialised.
To do this, you can generate a requirements file, like so:

```
cd <project_location>
source venv/bin/activate
pip3 freeze > requirements.txt
```

The above install process, then becomes:

```
cd <project_location>
python3 -m venv venv
source venv/bin/activate
python3 get-pip.py
pip3 install -r requirements.txt
```


I don't use a Mac, so I have had to modify some of the steps there. 

You may or may not need the` python3 get-pip.py`. I have found that I do need it on Linux. I don't think I had to include that on Windows.

**Reply by ghost** [reply]
I'll definitely look using a virtual environment myself as well as adding it to the Installation instructions. Thank you for your help and suggestions! :)

**Comment by TomSchimansky** [comment]
Looks good, you need to add screenshots to the repository to show people what to expect and get them to try it out.

**Reply by avalon60** [reply]
Yes, I was hoping to see what it would look like.
I got an error when trying to clone it:

```
git clone https://github.com/dishb/hive.git ./hive/
Cloning into './hive'...
fatal: invalid branch name: init.defaultBranch = 
```

**Reply by avalon60** [reply]
Fixed it with:
`
git config --global init.defaultBranch main`

**Reply by ghost** [reply]
@avalon60 - Yeah I was about to say I didn't see that `git` error on my machine!

**Reply by ghost** [reply]
@TomSchimansky - Thank you! I'll add some screenshots of the project in action today.

**Reply by avalon60** [reply]
And the bad news, for me is:

```
python3 ./utility/build.py
You are on an incompatible operating system. hive was designed for and tested on macOS.
```

Looks like you have to have a posh computer ;o)

**Reply by ghost** [reply]
@avalon60 - Haha, sorry! I'll work on making a Windows option as soon as I finish a few basic, necessary features. If you're on Linux, I think the program would work as Linux and macOS both use the same path-style.

If Linux, try disabling the `if` block on line 30, in `./utility/build.py`

**Reply by avalon60** [reply]
I'll need to do some work on it as I did actually try that, but got a different error. I'm eager to see what it looks like. It sounds pretty cool!

**Reply by ghost** [reply]
@avalon60 - Oh what- could you post the traceback? Perhaps its my code raising the Exception...

**Reply by ghost** [reply]
If it is my code - please raise an issue, thanks!

**Reply by avalon60** [reply]
Issue raised.

**Reply by ghost** [reply]
@avalon60 - Issue has been fixed :)

**Comment by ghost** [comment]
Added images! Now I'll set up the Discussions tab and hopefully start working on important issues :)

**Comment by avalon60** [comment]
It would be great if someone could get fontawesome icons working with CustomTkinter on Mac. They are amazing and take away much of the pain involved when blending icon colours with different themes. You can colour them, in much the same way as you can fonts. I wrote an app using the fontawesome module, and it worked great on Linux and Windows, but the library wouldn't install on Mac for some reason. It was a colleague who had the Mac and so it was difficult to look at the issue and try to resolve it.

@dishb, your hive project would really benefit. The icon library is very big and they are also consistent in appearance. For instance, there are picture icons. If you get as far as detecting file types, you could use such icons.

**Comment by ghost** [comment]
I think this is a great idea! I'd love to continue discussing this idea [here](https://github.com/dishb/hive/discussions/4).
